internal/schemautil/user_config/service_types.go

```diff
@@ -0,0 +1,1419 @@
+		},
+		"public_access": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"recovery_basebackup_name": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeString,
+		},
+		"service_to_fork_from": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+		"static_ips": {
+			Description: "Use static public IP addresses",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeBool,
+		},
+	}
+
+	return &schema.Schema{
+		Description:      "Influxdb user configurable settings",
+		DiffSuppressFunc: schemautil.EmptyObjectDiffSuppressFuncSkipArrays(s),
+		Elem:             &schema.Resource{Schema: s},
+		MaxItems:         1,
+		Optional:         true,
+		Type:             schema.TypeList,
+	}
+}
+
+// ServiceTypeKafkaMirrormaker is a generated function returning the schema of the kafka_mirrormaker ServiceType.
+func ServiceTypeKafkaMirrormaker() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"ip_filter_map": {
+			Description: "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeMap,
+		},
+		"ip_filter_string": {
+			Description: "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"kafka_mirrormaker": {
+			Optional: false,
+			Required: true,
+			Type:     schema.TypeMap,
+		},
+		"static_ips": {
+			Description: "Use static public IP addresses",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeBool,
+		},
+	}
+
+	return &schema.Schema{
+		Description:      "KafkaMirrormaker user configurable settings",
+		DiffSuppressFunc: schemautil.EmptyObjectDiffSuppressFuncSkipArrays(s),
+		Elem:             &schema.Resource{Schema: s},
+		MaxItems:         1,
+		Optional:         true,
+		Type:             schema.TypeList,
+	}
+}
+
+// ServiceTypePg is a generated function returning the schema of the pg ServiceType.
+func ServiceTypePg() *schema.Schema {
+	s := map[string]*schema.Schema{
+		"admin_password": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+		"admin_username": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeString,
+		},
+		"backup_hour": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeInt,
+		},
+		"backup_minute": {
+			Optional: true,
+			Required: false,
+			Type:     schema.TypeInt,
+		},
+		"enable_ipv6": {
+			Description: "Register AAAA DNS records for the service, and allow IPv6 packets to service ports",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeBool,
+		},
+		"ip_filter_map": {
+			Description: "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
+			Optional:    false,
+			Required:    true,
+			Type:        schema.TypeMap,
+		},
+		"ip_filter_string": {
```

Comment:
It is best we keep an existing fields like `ip_filter` intact, without the `_string` postfix.

Reply:
we already discussed this internally with you and yes, I will make it this way when the final version of this PR is ready